### ISSUE TITLE
Rename "master" branch to "main" for all our repos

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/community#7]
Many projects are renaming their default branch from `master` to `main` for good reasons, git and GitHub now also default to that name, and GitHub provided some tools to make this renaming easy (https://github.com/github/renaming). So let's do that for our repos in the geopandas org as well?

Some newer repos already use main (https://github.com/geopandas/xyzservices, https://github.com/geopandas/benchmarks, https://github.com/geopandas/community, https://github.com/geopandas/geopandas-user-surveys).  
Others should be renamed:

- [x] https://github.com/geopandas/geopandas
- [x] https://github.com/geopandas/contextily (cc @darribas)
- [x] https://github.com/geopandas/dask-geopandas (cc @jsignell)
- [x] https://github.com/geopandas/pyogrio (cc @brendan-ward)
- [x] https://github.com/geopandas/geo-arrow-spec

Workflow:
- Rename on GitHub: the renaming of the branch itself is aided by GitHub, see https://github.com/github/renaming#renaming-existing-branches
- Update branch name in code: for each repo, we should check occurrences of master in the code and rename those as well (in github actions, developer docs, etc)
- Update readthedocs settings

Other things?

---

## For contributors

The GitHub interface will typically explain what you need to do as contributor to a project that changed the default name, but putting it here as well (copying the instructions from dask https://github.com/dask/dask/issues/7199).

### What you'll see

Once the name on github is changed, when you try to `git pull` you'll get

```shell
Your configuration specifies to merge with the ref 'refs/heads/master'
from the remote, but no such ref was fetched.
```

### What you need to do

First: **head to your fork and rename the default branch there** Then:

```shell
git branch -m master main
git fetch origin
git branch -u origin/main main
```

**Comment by martinfleis** [comment]
I was doing that elsewhere not a long time ago and the process is relatively frictionless, esp. if the CI is on GHA. 

> Other things?

Change the default branch in `codecov` settings.

**Comment by jorisvandenbossche** [comment]
OK, I further updated all remaining projects. I also updated the readthedocs settings for all of them, I think (geopandas, contextily, dask-geopandas, pyogrio). 
And also updated codecov default branch setting for geopandas, contextily and dask-geopandas (pyogrio has no codecov I think). 

Something not included in the instructions above, and that might be relevant for every contributor: I also renamed the default branch in my forks (which doesn't seem to happen automatically). Not sure how important this is though, as you basically never use the main/master branch there ..
